cpp/unreal_plugins/SpEngine/Source/SpEngine/GameWorldService.h

```diff
@@ -6,27 +6,186 @@
+    LOAD_RegenerateBulkDataGuids     = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_RegenerateBulkDataGuids),
+    LOAD_DisableDependencyPreloading = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_DisableDependencyPreloading),
+    LOAD_Quiet                       = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_Quiet),
+    LOAD_FindIfFail                  = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_FindIfFail),
+    LOAD_MemoryReader                = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_MemoryReader),
+    LOAD_NoRedirects                 = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_NoRedirects),
+    LOAD_ForDiff                     = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_ForDiff),
+    LOAD_PackageForPIE               = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_PackageForPIE),
+    LOAD_DeferDependencyLoads        = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_DeferDependencyLoads),
+    LOAD_ForFileDiff                 = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_ForFileDiff),
+    LOAD_DisableCompileOnLoad        = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_DisableCompileOnLoad),
+    LOAD_DisableEngineVersionChecks  = static_cast<std::underlying_type_t<ELoadFlags>>(ELoadFlags::LOAD_DisableEngineVersionChecks)
+};
+
+// These USTRUCT types are intended to be wrappers for the UENUM types declared above. Wrapping enums in
+// structs like this helps us take advantage of UnrealObj and UnrealObjUtils to pass enums to and from
+// Python as a JSON string.
+
+USTRUCT()
+struct FSpIncludeSuperFlag
+{
+    GENERATED_BODY()
+
+    UPROPERTY()
+    ESpIncludeSuperFlag Enum;
+};
+
+USTRUCT()
+struct FSpObjectFlags
+{
+    GENERATED_BODY()
+
+    UPROPERTY()
+    ESpObjectFlags Enum;
+};
+
+USTRUCT()
+struct FSpLoadFlags
+{
+    GENERATED_BODY()
+
+    UPROPERTY()
+    ESpLoadFlags Enum;
+};
+
+// This struct is intended to be identical to Unreal's FActorSpawnParameters struct, see Engine/Source/Runtime/Engine/Classes/Engine/World.h
+USTRUCT()
+struct FSpActorSpawnParameters
+{
+    GENERATED_BODY()
+
+    UPROPERTY()
+    FName Name = NAME_None;
+
+    UPROPERTY()
+    AActor* Template = nullptr;
+
+    UPROPERTY()
+    AActor* Owner = nullptr;
+
+    UPROPERTY()
+    APawn* Instigator = nullptr;
+
+    UPROPERTY()
+    ULevel* OverrideLevel = nullptr;
+
+    UPROPERTY()
+    UChildActorComponent* OverrideParentComponent = nullptr;
+
+    UPROPERTY()
+    ESpawnActorCollisionHandlingMethod SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::Undefined;
+
+    UPROPERTY()
+    ESpawnActorScaleMethod TransformScaleMethod = ESpawnActorScaleMethod::MultiplyWithRoot;
+
+    UPROPERTY()
+    uint8 bNoFail = false;
```

Comment:
It's weird that we're assigning `false` to a `uint8`. And it's weird that it's a `uint8` but it has a `b` prefix in its name. Shouldn't we make this variable a `bool`?

Reply:
Ah yes, this can be made a bool as it's much easier to read. Unreal sometimes uses `uint8 bVar:1` instead of `bool bVar`.